distributed/dashboard/components/scheduler.py

```diff
@@ -343,12 +342,9 @@ def update(self):
                 "unmanaged_recent": [meminfo.unmanaged_recent] * 4,
                 "spilled": [meminfo.managed_spilled] * 4,
             }
-            # FIXME https://github.com/dask/distributed/issues/4675
-            #       This causes flickering after adding workers and when enough memory
-            #       is spilled out
-            self.root.x_range.end = max(
-                limit, meminfo.process + meminfo.managed_spilled
-            )
+
+            x_end = max(limit, meminfo.process + meminfo.managed_spilled)
+            self.root.x_range = DataRange1d(start=0, end=x_end, range_padding=0)
```

Comment:
Is this equivalent to creating a single `DataRange1d` instance in the `__init__` method and then updating `self.root.x_range.end` here?  I ask because it's seems to be about 10x faster to update an existing `DataRange1d` instance than creating a new one every time `update` is called

```python
In [1]: from bokeh.models import DataRange1d

In [2]: %timeit DataRange1d(start=0, end=100, range_padding=0)
80.6 µs ± 846 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [3]: x = DataRange1d(start=0, end=100, range_padding=0)

In [4]: %timeit x.end = 200
8.52 µs ± 71.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
That was a similar approach to what was implemented before. Where we weren't using `DataRange1D` but we were only updating the `x.end`.  I've just tried doing exactly what you suggest and this causes jittering as we had before : ( .

Reply:
A 10x slow down sounds bad.  A 70us slowdown sounds not so bad :)

Reply:
> A 10x slow down sounds bad. A 70us slowdown sounds not so bad :)

Yep, agreed. I was hoping that we could write very similar code, which was a bit more performant since this is called every 100ms on the scheduler. Though from @ncclementi it sounds like these two approaches actually aren't equivalent, so let's go with the current set of changes here